lib/spack/spack/filesystem_view.py

```diff
@@ -210,9 +210,16 @@ def __init__(self, root, layout, **kwargs):
             with open(projections_path, 'w') as f:
                 f.write(s_yaml.dump({'projections': self.projections}))
         else:
-            msg = 'View at %s has projections file' % self._root
-            msg += ' and was passed projections manually.'
-            raise ConflictingProjectionsError(msg)
+            # Ensure projections are the same from each source
+            # Read projections file from view
+            with open(projections_path, 'r') as f:
+                projections_data = s_yaml.load(f)
+                spack.config.validate(projections_data,
+                                      spack.schema.projections.schema)
+                if self.projections != projections_data['projections']:
+                    msg = 'View at %s has projections file' % self._root
+                    msg += ' which does not match projections passed manually.'
+                    raise ConflictingProjectionsError(msg)
```

Comment:
Should we just make environments the only way to make a view with projections?  Is that going to make people mad?

Reply:
I think that would make people mad. I know people are using separate environments to specify sets of package preferences that are then combined to form one user environment which could require a combinatorial projection.